lib/galaxy/datatypes/sniff.py

```diff
@@ -56,6 +56,9 @@ def stream_url_to_file(path, file_sources=None):
         file_source_path.file_source.realize_to(file_source_path.path, temp_name)
         return temp_name
     else:
+        if 'file://' in path:
+            # This shouldn't happen, but seems like a reasonable safeguard to prevent exposing system secrets
+            raise ValueError(f"stream_url_to_file received file:// URL '{path}', this is not allowed")
```

Comment:
I think this check needs to be removed, `file://` is allowed for dataset uploads by admin users when `allow_path_paste` is `True`, see test/integration/test_upload_configuration_options.py (which containts some integration tests that are broken by this).

```suggestion
```

Reply:
Lets try

Reply:
I disagree, this should be handled the same way as data_fetch does, which is to run through the appropriate checks and then pass just the path on. I would have thought that is what happens in https://github.com/mvdbeek/galaxy/blob/7febfd345fe242f92d652371400524028d1e9fc6/lib/galaxy/tools/parameters/grouping.py#L382, but if that is not the case we should check where that happens and do that as early as possible when the request comes in,

Reply:
I see, that's sort of intentional, streaming the source file to another location. https://github.com/mvdbeek/galaxy/blob/7febfd345fe242f92d652371400524028d1e9fc6/lib/galaxy/tools/parameters/grouping.py#L382 passes the file:// url on. It worries me a little that we may end up accidentally allowing `file://` uploads, but I guess that's why we have reviews :/.